test/lib/runner/mocha-runner/skip/index.js

```
@@ -1,51 +1,149 @@
 'use strict';
 
+const _ = require('lodash');
 const Skip = require('../../../../../lib/runner/mocha-runner/skip/');
 
 describe('Skip', () => {
-    let skip, test;
+    let skip;
+
+    const mkTest = (opts) => {
+        return _.defaults(opts || {}, {
+            parent: {
+                suites: [],
+                tests: []
+            }
+        });
+    };
 
     beforeEach(() => {
         skip = new Skip();
-        test = {};
     });
 
     describe('constructor', () => {
-        it('should not skip test by default', () => {
+        it('should not skip test silently by default', () => {
             assert.equal(skip.shouldSkip, false);
+            assert.equal(skip.silent, false);
         });
     });
 
     describe('handleEntity', () => {
         it('should extend test data with additional info', () => {
+            const test = mkTest();
             skip.shouldSkip = true;
+            skip.silent = false;
+
             skip.handleEntity(test);
 
             assert.propertyVal(test, 'pending', true);
             assert.property(test, 'skipReason');
         });
 
+        it('should not extend test data if test should be skipped silently', () => {
```

описание тестов какие-то технические.
`should not mark silently skipped test as skipped`